pkg/gofr/datasource/pubsub/mqtt/mqtt.go

```diff
@@ -338,19 +342,41 @@ func (m *MQTT) Unsubscribe(topic string) error {
 		close(sub.msgs)
 		delete(m.subscriptions, topic)
 	}
-	m.mu.RUnlock()
 
 	return nil
 }
 
-func (m *MQTT) Disconnect(waitTime uint) {
+func (m *MQTT) Close() error {
+	if m.config.CloseTimeout != 0 {
+		m.Client.Disconnect(uint(m.config.CloseTimeout.Milliseconds()))
+
+		return nil
+	}
+
+	// setting default value if value is not populated
+	m.Client.Disconnect(uint(m.config.CloseTimeout.Milliseconds()))
+
+	return nil
+}
```

Comment:
Here is what I would suggest


```suggestion
func (m *MQTT) Close() error {

	timeout := m.config.CloseTimeout

	if timeout == 0 {
	    // use default if unset
		  timeout = CloseTimeout
	}
	m.Client.Disconnect(uint(timeout.Milliseconds()))

	return nil
}
```

Reply:
Ok seems better. Will implement in this way